tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/PostFundingResource.java

```diff
@@ -43,11 +47,21 @@ public class PostFundingResource {
   private final Logger log = LoggerFactory.getLogger(PostFundingResource.class);
   private final PostFundingService postFundingService;
   private final PostFundingValidator postFundingValidator;
+  private final ApplicationEventPublisher applicationEventPublisher;
 
+  /**
+   * Constructor.
+   *
+   * @param postFundingService        service
+   * @param postFundingValidator      mapper
+   * @param applicationEventPublisher event publisher
+   */
```

Comment:
The code smells are there to encourage good practice.  We should go beyond satisfying the computer validated requirement.  Also the validator isn't a mapper :wink:.

 ```java
  /**
   * Constructs a PostFundingResource with required dependencies.
   *
   * @param postFundingService        encapsulates operations including persistence
   * @param postFundingValidator      validates {@link PostFundingDTO}s, including reference data
   * @param applicationEventPublisher publishes events for changes to a PostFunding
   */
```

Reply:
That'll teach me to think I can copy and paste and get away with it 😁